### PR TITLE
code coverage

### PR DESCRIPTION
**Pull request by MichaelChirico**
Still a bunch to fix in the next few days

### CODE REVIEW

File: R/data.table.R [hunk 1]
```diff
@@ -823,6 +836,7 @@ chmatch2 <- function(x, table, nomatch=NA_integer_) {
       } else {
         stop("When with=FALSE, j-argument should be of type logical/character/integer indicating the columns to select.") # fix for #1440.
       }
+>>>>>>> 7a662249f1d7755179ce7ec2cddc5c3afdb6c951
```
Comment (jangorecki) [inline R/data.table.R]: standard procedure after each merge conflict, ctrl+f: `>>>`, `<<<`, `===`

Comment (MichaelChirico) [reply R/data.table.R]: yes, noticed this, but there was a conflicting logic change, need to spend a few minutes to make sure the merge is correct... thanks for the eyes already :)

**Comment by MichaelChirico** [comment]
@jangorecki @mattdowle should be fixed now, ready to merge

**Comment by MichaelChirico** [comment]
Back to WIP... there are a few comments I left for myself to investigate and/or file issues:

https://github.com/Rdatatable/data.table/pull/3260/files#diff-595e663fedb004ffced209a139805080R9800

**Comment by MichaelChirico** [comment]
@mattdowle OK I think this is ready to merge

**Pull request merged**